subsys/net/lib/aws_iot/Kconfig

```diff
@@ -108,7 +108,6 @@ config AWS_IOT_CONNECTION_POLL_THREAD
 
 config AWS_IOT_TLS_SESSION_CACHING
```

Comment:
Why keep this option if it is not supported at all?
AWS has not offered this [for years](https://forums.aws.amazon.com/thread.jspa?threadID=219329) and is not likely to add it.

Reply:
Interesting, I read the answers from AWS differently, as if they're actually working on a solution for this, albeit very slowly.
The options are originally there because I wanted to have support for the feature in an official NCS release ready to use from the day AWS and Azure potentially add support for it, and not let users wait for next release. 
But it's not very important, so I'll remove the code.

Reply:
I would suspect that the pain of not having such features is growing every day?

Reply:
Would be great if they could add this, most likely in the form of session tickets if it were to happen.
Currently users must choose either to spend quite a bit of power and data to send MQTT keepalive every 15-20 min or disconnect and do a very expensive TLS handshake everytime they want to talk to cloud.